include/irr/asset/NblAssetCompileConfig.h

```diff
@@ -0,0 +1,204 @@
+#define NABLA_VERSION_MINOR 3
+#define NABLA_VERSION_REVISION 0
+// This flag will be defined only in SVN, the official release code will have
+// it undefined
+//#define IRRLICHT_VERSION_SVN -alpha
+#define NABLA_SDK_VERSION "0.3.0-beta2"
+
+#define NEW_MESHES
+
+#include <stdio.h> // TODO: Although included elsewhere this is required at least for mingw
+
+//! Passing defines to the compiler which have NO in front of the _NBL definename is an alternative
+//! way which can be used to disable defines (instead of outcommenting them in this header).
+//! So defines can be controlled from Makefiles or Projectfiles which allows building
+//! different library versions without having to change the sources.
+//! Example: NO_NBL_COMPILE_WITH_X11_ would disable X11
+
+// this actually includes file depending on build type (Debug/Release)
+#include "BuildConfigOptions.h"
+
+#define NEW_SHADERS 1
+
+#ifdef _MSC_VER
+#	define _ENABLE_EXTENDED_ALIGNED_STORAGE
+#endif
+
+//! VidMode is ANCIENT
+//#define NO_NBL_LINUX_X11_VIDMODE_
+
+//! On some Linux systems the XF86 vidmode extension or X11 RandR are missing. Use these flags
+//! to remove the dependencies such that Irrlicht will compile on those systems, too.
+//! If you don't need colored cursors you can also disable the Xcursor extension
+#if defined(_NBL_PLATFORM_LINUX_) && defined(_NBL_COMPILE_WITH_X11_)
+#   define _NBL_LINUX_X11_VIDMODE_
+#   define _NBL_LINUX_X11_RANDR_
+#   ifdef NO_NBL_LINUX_X11_VIDMODE_
+#       undef _NBL_LINUX_X11_VIDMODE_
+#   endif
+#   ifdef NO_NBL_LINUX_X11_RANDR_
+#       undef _NBL_LINUX_X11_RANDR_
+#   endif
+#endif
+
+//! Define _NBL_COMPILE_WITH_X11_ to compile the Irrlicht engine with X11 support.
+/** If you do not wish the engine to be compiled with X11, comment this
+define out. */
+// Only used in LinuxDevice.
+///#ifndef _NBL_SERVER_
+#define _NBL_COMPILE_WITH_X11_
+///#endif
+#ifdef NO_NBL_COMPILE_WITH_X11_
+#   undef _NBL_COMPILE_WITH_X11_
+#endif
+
+//! Define _NBL_COMPILE_WITH_OPENSSL_ to enable compiling the engine using libjpeg.
+/** This enables the engine to read and write encrypted BAW format files.
+If you comment this out, the engine will no longer read or write encrypted .baw files! */
+#define _NBL_COMPILE_WITH_OPENSSL_
+#ifdef NO_NBL_COMPILE_WITH_OPENSSL_
+#undef _NBL_COMPILE_WITH_OPENSSL_
+#endif
+
+//! Define _NBL_COMPILE_WITH_JPEGLIB_ to enable compiling the engine using libjpeg.
+/** This enables the engine to read jpeg images. If you comment this out,
+the engine will no longer read .jpeg images. */
+#define _NBL_COMPILE_WITH_LIBJPEG_
+#ifdef NO_NBL_COMPILE_WITH_LIBJPEG_
+#undef _NBL_COMPILE_WITH_LIBJPEG_
+#endif
+
+//! Define _NBL_COMPILE_WITH_LIBPNG_ to enable compiling the engine using libpng.
+/** This enables the engine to read png images. If you comment this out,
+the engine will no longer read .png images. */
+#define _NBL_COMPILE_WITH_LIBPNG_
+#ifdef NO_NBL_COMPILE_WITH_LIBPNG_
+#undef _NBL_COMPILE_WITH_LIBPNG_
+#endif
```

Comment:
this goes stays in `asset` also the first comment is wrong, its for compiling with libssl